statsmodels/genmod/families/family.py

```diff
@@ -1275,14 +1279,9 @@ def loglike(self, endog, lin_pred=None):
            constant = gammaln(endog + 1/alpha) - gammaln(endog + 1) -
                       gammaln(1/alpha)
         """
-        # don't need to specify mu
-        if lin_pred is None:
-            raise AttributeError('The loglikelihood for the negative binomial'
-                                 ' requires that the fitted values be '
-                                 'provided via the `lin_pred` keyword '
-                                 'argument.')
-        constant = (special.gammaln(endog + 1/self.alpha) -
-                    special.gammaln(endog+1) - special.gammaln(1/self.alpha))
+        lin_pred = self._link(mu)
+        constant = special.gammaln(endog + 1/self.alpha) - special.gammaln(endog+1)\
+                    -special.gammaln(1/self.alpha)
         exp_lin_pred = np.exp(lin_pred)
```

Comment:
Is there a bug .here?  Why do we take exp of lin_pred instead of `link.inverse(linpred)` i.e. mu
looks to me the loglike assumes log link and doesn't depend on actual link

I haven't verified, this just looks suspicious to me while reading. And it's not in this PR